elastic/kibana
# [eem] definition update endpoint

Create `PATCH kbn:/internal/entities/definition` allowing update of stored definition. The endpoint accepts an update object representing a partial entity definition with a few key properties that cannot be updated. The update process will stop and delete the existing transforms and create new ones, there's https://github.com/elastic/elastic-entity-model/issues/136 logged as a follow up improvement.

### Testing
- call `PUT kbn:/internal/entities/definition/{id}` with an update payload (see [update schema](https://github.com/elastic/kibana/blob/7a7fbdf1cd5aba9318465fd34414e7e6fa73f6b7/x-pack/packages/kbn-entities-schema/src/schema/entity_definition.ts#L62))
- call `GET kbn:/internal/entities/definition` and verify the update in reflected in the stored definition
- verify the updated properties are reflected in the corresponding transform/pipeline components
